src/test/java/gim/model/date/DateTest.java

```diff
@@ -1,26 +1,63 @@
 package gim.model.date;
 
 import static gim.testutil.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
 
 import org.junit.jupiter.api.Test;
 
 public class DateTest {
 
+    /**
+     * Date constructor does not accept null as its value. When no
+     * argument is provided to the constructor, the value is set by
+     * default as today's date.
+     */
+    @Test
+    public void constructor_default_setAsDateToday() {
+        assertEquals((new Date()).date, LocalDate.now());
+    }
+
     @Test
     public void constructor_null_throwsNullPointerException() {
         assertThrows(NullPointerException.class, () -> new Date(null));
     }
 
+    /**
+     * Test for invalid date where the formatting does not conform to dd/MM/uuuu format.
+     */
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDateEmptyString = "";
```

Comment:
Maybe test for a string of spaces  "            " as well?

Reply:
The test has been updated!